datalad/interface/rerun.py

```diff
@@ -0,0 +1,300 @@
+
+        do_checkout = branch
+        if onto is not None:
+            if onto.strip() == "":
+                ## An empty argument means go to the parent of the
+                ## first revision, but that doesn't exist for
+                ## --since=.  Instead check out an orphan branch.
+                if root and branch:
+                    ds.repo.checkout(branch, options=["--orphan"])
+                    # Make sure we are actually on an orphan branch
+                    # before doing a hard reset.
+                    if ds.repo.get_hexsha():
+                        yield dict(
+                            err_info, status="error",
+                            message="failed to create orphan branch")
+                        return
+                    ds.repo.repo.git.reset("--hard")
+                    do_checkout = False
+                elif root:
+                    yield dict(
+                        err_info, status="error",
+                        message="branch name is required for orphan")
+                    return
+                else:
+                    ds.repo.checkout(revs[0] + "^", options=["--detach"])
+            else:
+                ds.repo.checkout(onto, options=["--detach"])
+        if do_checkout:
+            ds.repo.checkout("HEAD", ["-b", branch])
+
+        for rev in revs:
+            # pull run info out of the revision's commit message
+            try:
+                rec_msg, runinfo = get_commit_runinfo(ds.repo, rev)
+            except ValueError as exc:
+                yield dict(
+                    err_info, status='error',
+                    message=str(exc)
+                )
+                return
+            if not runinfo:
+                pick = False
+                try:
+                    ds.repo.repo.git.merge_base("--is-ancestor", rev, "HEAD")
+                except GitCommandError:  # Revision is NOT an ancestor of HEAD.
+                    pick = True
+
+                shortrev = ds.repo.repo.git.rev_parse("--short", rev)
+                err_msg = "no command for {} found; {}".format(
+                    shortrev,
+                    "cherry picking" if pick else "skipping")
+                yield dict(err_info, status='ok', message=err_msg)
+
+                if pick:
+                    ds.repo.repo.git.cherry_pick(rev)
+                continue
+
+            # now we have to find out what was modified during the
+            # last run, and enable re-modification ideally, we would
+            # bring back the entire state of the tree with #1424, but
+            # we limit ourself to file addition/not-in-place-modification
+            # for now
+            for r in ds.unlock(new_or_modified(ds, rev),
+                               return_type='generator', result_xfm=None):
+                yield r
+
+            for r in run_command(runinfo['cmd'], ds, rec_msg or message,
```

Comment:
I wonder what the role of `--message` for rerun should be?
- here it is for the case if original commit message was not specified and we provided one to use instead (for every such commit).  Not sure what is the value of that 
- could be `message or rec_msg`. The reason I might want to specify a custom message is to record my intent ("Rerunning analysis in the new environment X").  It would make little sense then to use it only if original commit message had no message and otherwise ignore my "intent" record.   But it also makes little sense to use the same message for all commits if a range is specified.  May be we should allow for a "format" template may be? e.g. `--message="{}\n\nWas reran in env X"` or may be even exposing more details of the original commit? `{message}\n\nOriginal analysis which was done by {author} on {date} in commit {hexsha}`?

For either of those cases, we would need to customize the doc string for the `--message` option probably

Reply:
> could be message or rec_msg

eh, thanks for catching that.  That was the intent, but somewhere
along the way I flipped it :/  I'll correct it and add a test for
that.

The way I viewed message was mostly useful for rerunning a single
commit.  In the case rerunning several commits, it was supposed to
override every message, which is of course less useful, as you say
above.

re template: Not sure.  IMO we should wait to see if there is a need
for it.  For now, I'll plan to update the help message to describe the
"one message for all" behavior.